### PR TITLE
fix: jpeg segment underflow

### PR DESCRIPTION
**Pull request by cdmurph32**
Prevent panic from malformed jpeg.

### CODE REVIEW

File: src/jpeg/segment.rs [hunk 1]
```diff
@@ -195,3 +202,19 @@ impl fmt::Debug for JpegSegment {
...
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bytes::Bytes;
+
+    #[test]
+    fn test_encode_at_underflow() {
+        let segment =
+            JpegSegment::new_with_entropy(0xFF, Vec::new().into(), Bytes::from(vec![4, 5, 6]));
```
Comment (paolobarbolini) [inline src/jpeg/segment.rs:214]:
```suggestion
            JpegSegment::new_with_entropy(0xFF, Bytes::new(), Bytes::from(vec![4, 5, 6]));
```

File: src/jpeg/segment.rs [hunk 2]
```diff
@@ -177,7 +177,14 @@ impl EncodeAt for JpegSegment {
...
             _ => {
-                *pos -= 1 + !self.contents.is_empty() as usize + !self.entropy.is_empty() as usize;
+                let decrement =
+                    1 + !self.contents.is_empty() as usize + !self.entropy.is_empty() as usize;
+                if *pos >= decrement {
+                    *pos -= decrement;
+                } else {
+                    // Prevent underflow
+                    *pos = 0;
+                }
```
Comment (paolobarbolini) [inline src/jpeg/segment.rs:187]: Should be more idiomatic

```suggestion
                // Prevent underflow with malformed JPEGs
                *pos = pos.saturating_sub(decrement);
```

**Review by paolobarbolini — COMMENTED** [review]
Thanks for the PR. I've left a few nit picks

File: src/jpeg/segment.rs (hunk 1 above)
Comment (cdmurph32) [reply src/jpeg/segment.rs]: 👍

File: src/jpeg/segment.rs (hunk 2 above)
Comment (cdmurph32) [reply src/jpeg/segment.rs]: 👍

**Comment by paolobarbolini** [comment]
Released as [v0.3.2](https://crates.io/crates/img-parts/0.3.2)

**Pull request merged**